### PR TITLE
LINK-2042 |  Fields to select registration specific account and merchant

### PR DESCRIPTION
**Pull request by jorilindell**
## Description
Add fields to select a merchant and an account in the registration form. Display table to preview default values of the merchant or the account when an account/merchant is selected.
Allow to override default account values.

## Closes
[LINK-2042](https://helsinkisolutionoffice.atlassian.net/browse/LINK-2042)

## Screenshots
Screenshot 2024-06-03 at 14 14 59

Screenshot 2024-06-03 at 14 14 48

[LINK-2042]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-2042?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr400-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __failed__ for https://linkedcomponents-pr400-ui.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr400-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedcomponents-pr400-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr400-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __failed__ for https://linkedcomponents-pr400-ui.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr400-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __failed__ for https://linkedcomponents-pr400-ui.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedcomponents-pr400-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr400-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __failed__ for https://linkedcomponents-pr400-ui.dev.hel.ninja :crying_cat_face::anger::boom::boom:

File: src/domain/app/i18n/fi.json [hunk 1]
```diff
@@ -1883,6 +1882,39 @@
       "placeholderConfirmationMessage": "Syötä vahvistusviesti ilmoittautuneille",
       "placeholderEvent": "Valitse tai hae tapahtuma",
       "placeholderInstructions": "Syötä ilmoittautumisohjeet",
+      "registrationAccount": {
+        "buttonReturnDefaults": "Palauta tilin oletusarvot",
+        "helperDefaultValue": "Oletusarvo: {{value}}",
+        "labelAccount": "Kirjanpitotili",
+        "labelBalanceProfitCenter": "Tasapainotettu tulosyksikkö",
```
Comment (harriris-vincit) [inline src/domain/app/i18n/fi.json:1889]:
```suggestion
        "labelBalanceProfitCenter": "Tasetilin tulosyksikkö",
```
This should probably say "Tasetilin tulosyksikkö" which is the term used in the PowerPoint presentation sent by Jarkko.

File: src/domain/app/i18n/fi.json [hunk 2]
```diff
@@ -1883,6 +1882,39 @@
...
+        "labelCompanyCode": "SAP-yrityskoodi",
```
Comment (harriris-vincit) [inline src/domain/app/i18n/fi.json:1890]:
```suggestion
        "labelCompanyCode": "SAP-yritystunnus",
```
Term used in Jarkko's PPTX.

File: src/domain/app/i18n/fi.json [hunk 3]
```diff
@@ -1883,6 +1882,39 @@
...
+        "labelInternalOrder": "Sisäinen tilaus",
+        "labelMainLedgerAccount": "Pääkirjanpitotili",
```
Comment (harriris-vincit) [inline src/domain/app/i18n/fi.json:1892]:
```suggestion
        "labelMainLedgerAccount": "Pääkirjatili",
```
Ledgerillä viitataan pääkirjaan eli General Ledgeriin.

File: src/domain/app/i18n/fi.json [hunk 4]
```diff
@@ -1883,6 +1882,39 @@
...
+        "labelOperationArea": "SAP-toimintoalue",
+        "labelProfitCenter": "Tulosyksikkö",
+        "labelProject": "Projekti",
+        "placeholderAccount": "Valitse kirjanpitotili",
+        "placeholderBalanceProfitCenter": "Syötä tasapainotettu tulosyksikkö",
```
Comment (harriris-vincit) [inline src/domain/app/i18n/fi.json:1897]:
```suggestion
        "placeholderBalanceProfitCenter": "Syötä taseen tulosyksikkö",
```

File: src/domain/app/i18n/fi.json [hunk 5]
```diff
@@ -1883,6 +1882,39 @@
...
+        "placeholderCompanyCode": "Syötä SAP-yrityskoodi",
```
Comment (harriris-vincit) [inline src/domain/app/i18n/fi.json:1898]:
```suggestion
        "placeholderCompanyCode": "Syötä SAP-yritystunnus",
```

File: src/domain/app/i18n/fi.json [hunk 6]
```diff
@@ -1883,6 +1882,39 @@
...
+        "placeholderInternalOrder": "Syötä sisäinen tilaus",
+        "placeholderMainLedgerAccount": "Syötä pääkirjanpitotili",
```
Comment (harriris-vincit) [inline src/domain/app/i18n/fi.json:1900]:
```suggestion
        "placeholderMainLedgerAccount": "Syötä pääkirjatili",
```

File: src/domain/app/i18n/sv.json [hunk 7]
```diff
@@ -1883,6 +1882,39 @@
       "placeholderConfirmationMessage": "Ange bekräftelsemeddelande för registranter",
       "placeholderEvent": "Välj eller sök efter evenemanget",
       "placeholderInstructions": "Ange instruktioner för registrering",
+      "registrationAccount": {
+        "buttonReturnDefaults": "Återställa kontots förinställda värden",
+        "helperDefaultValue": "Förinställt värde: {{value}}",
+        "labelAccount": "Redovisningskonto",
+        "labelBalanceProfitCenter": "Balansera resultatenheten",
```
Comment (harriris-vincit) [inline src/domain/app/i18n/sv.json:1889]:
```suggestion
        "labelBalanceProfitCenter": "Balansräkningens resultatenhet",
```

File: src/domain/app/i18n/sv.json [hunk 8]
```diff
@@ -1883,6 +1882,39 @@
...
+        "labelCompanyCode": "SAP företagskod",
+        "labelInternalOrder": "Intern ordning",
+        "labelMainLedgerAccount": "Huvudbokkonto",
+        "labelOperationArea": "SAP funktionsområde",
+        "labelProfitCenter": "Resultatenheten",
+        "labelProject": "Projekt",
+        "placeholderAccount": "Välj redovisningskonto",
+        "placeholderBalanceProfitCenter": "Ange balansera resultatenheten",
```
Comment (harriris-vincit) [inline src/domain/app/i18n/sv.json:1897]:
```suggestion
        "placeholderBalanceProfitCenter": "Ange balansräkningens resultatenhet",
```

File: src/domain/app/i18n/sv.json [hunk 9]
```diff
@@ -1883,6 +1882,39 @@
...
+        "placeholderCompanyCode": "Ange SAP företagskod",
+        "placeholderInternalOrder": "Ange intern ordning",
+        "placeholderMainLedgerAccount": "Ange huvudbokkonto",
+        "placeholderOperationArea": "Ange SAP funktionsområde",
+        "placeholderProfitCenter": "Ange resultatenheten",
+        "placeholderProject": "Ange projekt"
+      },
+      "registrationMerchant": {
+        "labelMerchant": "HandlareKauppias",
```
Comment (harriris-vincit) [inline src/domain/app/i18n/sv.json:1906]:
```suggestion
        "labelMerchant": "Handlare",
```

File: src/domain/organization/organizationForm/__tests__/OrganizationForm.test.tsx [hunk 10]
```diff
@@ -64,7 +64,7 @@ test('should add and remove account', async () => {
   await shouldAddAndRemoveItem({
     addButtonLabel: 'Lisää uusi tili',
     deleteButtonLabel: 'Poista tili',
-    expectedInputLabel: 'ALV-koodi *',
+    expectedInputLabel: 'SAP-yrityskoodi *',
```
Comment (harriris-vincit) [inline src/domain/organization/organizationForm/__tests__/OrganizationForm.test.tsx:67]:
```suggestion
    expectedInputLabel: 'SAP-yritystunnus *',
```

File: src/domain/registration/formSections/priceGroups/__tests__/PriceGroupSection.test.tsx [hunk 11]
```diff
@@ -51,6 +75,33 @@ const getElement = (
   }
 };
 
+const accountFields = [
+  { name: 'SAP-yrityskoodi *', value: accountPayload.companyCode },
```
Comment (harriris-vincit) [inline src/domain/registration/formSections/priceGroups/__tests__/PriceGroupSection.test.tsx:79]:
```suggestion
  { name: 'SAP-yritystunnus *', value: accountPayload.companyCode },
```

File: src/domain/registration/formSections/priceGroups/__tests__/PriceGroupSection.test.tsx [hunk 12]
```diff
@@ -51,6 +75,33 @@ const getElement = (
...
+  { name: 'Pääkirjanpitotili *', value: accountPayload.mainLedgerAccount },
```
Comment (harriris-vincit) [inline src/domain/registration/formSections/priceGroups/__tests__/PriceGroupSection.test.tsx:80]:
```suggestion
  { name: 'Pääkirjatili *', value: accountPayload.mainLedgerAccount },
```

File: src/domain/registration/formSections/priceGroups/__tests__/PriceGroupSection.test.tsx [hunk 13]
```diff
@@ -51,6 +75,33 @@ const getElement = (
...
+  {
+    name: 'Tasapainotettu tulosyksikkö *',
```
Comment (harriris-vincit) [inline src/domain/registration/formSections/priceGroups/__tests__/PriceGroupSection.test.tsx:82]:
```suggestion
    name: 'Taseen tulosyksikkö *',
```

File: src/domain/app/i18n/fi.json (hunk 1 above)
Comment (jorilindell) [reply src/domain/app/i18n/fi.json]: Thanks. I was quite sure that it's not "Tasapainotettu tulosyksikkö" which was offered by Google and Deepl. Changed now

File: src/domain/app/i18n/fi.json (hunk 2 above)
Comment (jorilindell) [reply src/domain/app/i18n/fi.json]: Changed as suggested

**Review by harriris-vincit — APPROVED** [review]
LGTM, and tested to work by creating and then modifying a registration using different merchant and account values.

File: src/domain/app/i18n/fi.json (hunk 4 above)
Comment (jorilindell) [reply src/domain/app/i18n/fi.json]: I changed this to "Syötä tasetilin tulosyksikkö" to be consistent with the field label text.

File: src/domain/app/i18n/fi.json (hunk 3 above)
Comment (jorilindell) [reply src/domain/app/i18n/fi.json]: Changed as suggested

File: src/domain/app/i18n/fi.json (hunk 5 above)
Comment (jorilindell) [reply src/domain/app/i18n/fi.json]: Changed as suggested

File: src/domain/app/i18n/fi.json (hunk 6 above)
Comment (jorilindell) [reply src/domain/app/i18n/fi.json]: Changed as suggested

File: src/domain/app/i18n/sv.json (hunk 7 above)
Comment (jorilindell) [reply src/domain/app/i18n/sv.json]: Changed as suggested

File: src/domain/app/i18n/sv.json (hunk 9 above)
Comment (jorilindell) [reply src/domain/app/i18n/sv.json]: What an irritation copy-paste error. Changed as suggested

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr400-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __failed__ for https://linkedcomponents-pr400-ui.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Pull request merged**